core/templates/dev/head/expressions/expressionSyntaxTree.js

```diff
@@ -0,0 +1,420 @@
+        throw new ExprWrongArgTypeError(
+          originalValue, typeof originalValue, 'Number');
+      };
+
+      // Coerces all values in the given argument array to Number, and throws
+      // an error if the result is NaN.
+      var _coerceAllArgsToNumber = function(args) {
+        for (var i = 0; i < args.length; i++) {
+          args[i] = _coerceToNumber(args[i]);
+        }
+        return args;
+      };
+
+      // NOTE TO DEVELOPERS: When adding a new reserved word to this object,
+      //   please first ensure that existing explorations do not use this
+      //   parameter name. Also, to prevent future explorations using it,
+      //   modify feconf.INVALID_PARAMETER_NAMES accordingly.
+      // TODO(kashida): Document all operators input and output contracts.
+
+      var system = {
+    '#': {
+      type: function(args, envs) {
+        return lookupEnvs(args + '', envs);
+      },
+      eval: function(args, envs) {
+        return lookupEnvs(args[0] + '', envs);
+      }
+    },
+    '+': {
+      type: function(args, envs) {
+        verifyNumArgs(args, 1, 2);
+        _verifyArgsHaveType(args, 'Real');
+        return 'Real';
+      },
+      eval: function(args, envs) {
+        verifyNumArgs(args, 1, 2);
+        var numericArgs = _coerceAllArgsToNumber(args);
+        return numericArgs.length == 1 ? numericArgs[0] :
+          numericArgs[0] + numericArgs[1];
+      }
+    },
+    '-': {
+      type: function(args, envs) {
+        verifyNumArgs(args, 1, 2);
+        _verifyArgsHaveType(args, 'Real');
+        return 'Real';
+      },
+      eval: function(args, envs) {
+        verifyNumArgs(args, 1, 2);
+        var numericArgs = _coerceAllArgsToNumber(args);
+        return numericArgs.length == 1 ? -numericArgs[0] :
+          numericArgs[0] - numericArgs[1];
+      }
+    },
+    '*': {
+      type: function(args, envs) {
+        verifyNumArgs(args, 2);
+        _verifyArgsHaveType(args, 'Real');
+        return 'Real';
+      },
+      eval: function(args, envs) {
+        verifyNumArgs(args, 2);
+        var numericArgs = _coerceAllArgsToNumber(args);
+        return numericArgs[0] * numericArgs[1];
+      }
+    },
+    '/': {
+      type: function(args, envs) {
+           verifyNumArgs(args, 2);
+           _verifyArgsHaveType(args, 'Real');
+           return 'Real';
+         },
+      eval: function(args, envs) {
+           verifyNumArgs(args, 2);
+           var numericArgs = _coerceAllArgsToNumber(args);
+           return numericArgs[0] / numericArgs[1];
+         }
+    },
+    '%': {
+      type: function(args, envs) {
+           verifyNumArgs(args, 2);
+           _verifyArgsHaveType(args, 'Real');
+           return 'Real';
+         },
+      eval: function(args, envs) {
+           verifyNumArgs(args, 2);
+           var numericArgs = _coerceAllArgsToNumber(args);
+           return numericArgs[0] % numericArgs[1];
+         }
+    },
+    '<=': {
+      type: function(args, envs) {
```

Comment:
Should this end up being boolean? OK if we don't do it now, but maybe add a todo